tokio/src/net/driver/readiness.rs

```diff
@@ -0,0 +1,311 @@
+
+/// A set of readiness event kinds.
+///
+/// `Readiness` is set of operation descriptors indicating which kind of an
+/// operation is ready to be performed.
+///
+/// This struct only represents portable event kinds. Portable events are
+/// events that can be raised on any platform while guaranteeing no false
+/// positives.
+///
+/// # Examples
+///
+/// ```rust
+/// use tokio::net::driver::Readiness;
+///
+/// let readiness = Readiness::readable() | Readiness::read_closed();
+/// assert!(readiness.is_readable());
+/// assert!(readiness.is_read_closed());
+/// ```
+#[derive(Clone, Copy, Debug, PartialEq, PartialOrd)]
+pub struct Readiness(usize);
+
+impl Readiness {
+    /// Returns the empty `Readiness` set.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::net::driver::Readiness;
+    ///
+    /// let readiness = Readiness::empty();
+    ///
+    /// assert!(!readiness.is_readable());
+    /// ```
+    pub fn empty() -> Readiness {
+        Readiness(0)
+    }
+
+    /// Returns a `Readiness` representing readable readiness.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::net::driver::Readiness;
+    ///
+    /// let readiness = Readiness::readable();
+    ///
+    /// assert!(readiness.is_readable());
+    /// ```
+    pub fn readable() -> Readiness {
+        Readiness(READABLE)
+    }
+
+    /// Returns a `Readiness` representing writable readiness.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::net::driver::Readiness;
+    ///
+    /// let readiness = Readiness::writable();
+    ///
+    /// assert!(readiness.is_writable());
+    /// ```
+    pub fn writable() -> Readiness {
+        Readiness(WRITABLE)
+    }
+
+    /// Returns a `Readiness` representing read closed readiness.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::net::driver::Readiness;
+    ///
+    /// let readiness = Readiness::read_closed();
+    ///
+    /// assert!(readiness.is_read_closed());
+    /// ```
+    pub fn read_closed() -> Readiness {
+        Readiness(READ_CLOSED)
+    }
+
+    /// Returns a `Readiness` representing write closed readiness.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::net::driver::Readiness;
+    ///
+    /// let readiness = Readiness::write_closed();
+    ///
+    /// assert!(readiness.is_write_closed());
+    /// ```
+    pub fn write_closed() -> Readiness {
+        Readiness(WRITE_CLOSED)
+    }
+
+    /// Returns a `Readiness` representing readiness for all operations.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::net::driver::Readiness;
+    ///
+    /// let readiness = Readiness::all();
+    ///
+    /// assert!(readiness.is_readable());
+    /// assert!(readiness.is_writable());
+    /// assert!(readiness.is_read_closed());
+    /// assert!(readiness.is_write_closed());
+    /// ```
+    pub fn all() -> Readiness {
+        Readiness(READABLE | WRITABLE | READ_CLOSED | WRITE_CLOSED)
+    }
```

Comment:
You may want to consider defining these as associated consts rather than functions. Something like this:
```rust
impl Readiness {
    pub const EMPTY: Self = Self(0);
    pub const READABLE: Self = Self(READABLE);
    pub const WRITABLE: Self = Self(WRITABLE);
    pub const READ_CLOSED: Self = Self(READ_CLOSED);
    pub const WRITE_CLOSED: Self = Self(WRITE_CLOSED);
    pub const ALL: Self = Self(READABLE | WRITABLE | READ_CLOSED | WRITE_CLOSED);
    // ...
}
```